crates/bevy_app/src/app.rs

```diff
@@ -921,5 +921,9 @@ fn run_once(mut app: App) {
 }
 
 /// An event that indicates the [`App`] should exit. This will fully exit the app process.
+///
+/// You can also use this event to detect that an exit was requested. In order to receive it, systems
+/// subscribing to this event should run before [`CoreStage::Last`], which is when the app exiting
+/// actually occurs.
```

Comment:
The last sentence needs validation, or might need to be removed if this is still unstable/subject to change.

Reply:
Yeah, it looks like my advice was wrong: https://github.com/bevyengine/bevy/blob/f907d67d7ed5cfd8b11b2ad78e9dfd6eeb913cea/crates/bevy_app/src/schedule_runner.rs#L96

https://github.com/bevyengine/bevy/blob/fb02b842244ae562a1e939a471ea6cc108e8c5f0/crates/bevy_winit/src/lib.rs#L591

In particular, this is handled by the runner, which means there's no timing trickery to think about. It will occur at the start of the next tick of the schedule.

Reply:
I can confirm; I tried adding the following system at CoreStage::Last and it did get the AppExit log:
 ```rust
 #[tracing::instrument(skip_all)]
fn test(exit: EventReader<AppExit>) {
    if !exit.is_empty() {
        warn!("AppExit");
    }
}
```
Neat!